psfws/gen_params.py

```diff
@@ -0,0 +1,371 @@
+        return cn2_complete, h_complete
+
+    def get_cn2_all(self):
+        """Get array of Cn2 values for all data available."""
+        cn2_list = []
+        for i in range(self.N):
+            cn2, h = self.get_cn2(i)
+            cn2_list.append(cn2)
+        return np.array(cn2_list), h
+
+    def _get_auto_layers(self):
+        """Return layer altitudes according to max wind speed & turbulence."""
+        # make an array of heights for interpolation
+        h_interp = np.linspace(self.h0 + self.h_dome, max(self.h_gfs), 500)
+
+        # interpolate the median speeds from GFS to find height of max
+        all_speeds = [i for i in self.gfs_winds['speed'].values]
+        h_maxspd = utils.find_max_median(all_speeds, self.h_gfs,
+                                         h_interp, self.h0)
+
+        # interpolate the median cn2 to find height of max
+        all_cn2, h_cn2 = self.get_cn2_all()
+        h_maxcn2 = utils.find_max_median(all_cn2, h_cn2, h_interp, self.h0)
+
+        # sort the heights of max speed and max turbulence
+        h3, h4 = np.sort([h_maxspd, h_maxcn2])
+
+        # raise the lowest layer slightly off of the ground
+        lowest = self.h0 + 0.250
+
+        h2 = np.mean([lowest, h3])
+        h5 = np.mean([h4, 18])
+
+        return [lowest, h2, h3, h4, h5, 18]
+
+    def _integrate_cn2(self, cn2, h, layers='auto'):
+        """Calculate turbulence integral of given Cn2 at given layers.
+
+        Parameters
+        ----------
+        cn2 : array
+            cn2 values
+        h : array
+            heights of cn2 values
+        layers : str or list or array
+            Centers of integration regions set by layers keyword; either
+            list/array of values, or 'auto' for them to be automatically
+            calculated based on wind and turbulence maximums (default: 'auto')
+
+        Returns
+        -------
+        turbulence integral values J, in m^(1/3)
+        integration bin edges, in km
+        integration bin centers (aka layers if not 'auto'), in km
+
+        """
+        maxh = max(self.h_gfs)
+
+        # define bins according to layers argument
+        if layers == 'auto':
+            bin_centers = self._get_auto_layers()
+        else:
+            bin_centers = layers
+        edges = [self.h0] + [np.mean(bin_centers[i:i+2])
+                             for i in range(len(bin_centers)-1)] + [maxh]
+
+        # integrate cn2 in bins defined by these edges
+        j = utils.integrate_in_bins(cn2, h, edges)
+        # return along with edges and bin centers
+        return j, np.array(edges), np.array(bin_centers)
+
+    def get_turbulence_integral(self, pt, layers='auto'):
+        """Return integrated Cn2 profile for datapoint with index pt."""
+        # get cn2 of pt
+        cn2, h = self.get_cn2(pt)
+
+        # return integrated cn2
+        return self._integrate_cn2(cn2, h, layers=layers)
+
+    def get_wind_interpolation(self, pt, h_out, kind='gp'):
```

Comment:
If you decide to have `kind` be a class attribute (see my comment at line 256), I think it would still make sense to have it be overwrittable here (if a user wants to play with different interpolations, for plotting purposes or to troubleshoot).

I would have kind default to `None` here, then in the method, add something like

```python
if kind is None:
    kind = self.interpolator